manifests/prepare.pp

```diff
@@ -53,15 +54,25 @@
 
   # TODO: manage server.cfg contents
   if $::mcollective_configured {
-    file { '/etc/puppetlabs/mcollective':
+    file { $::agent_upgrade::params::mcodirs:
       ensure => directory,
     }
-    file { '/etc/puppetlabs/mcollective/server.cfg':
+    file { $::agent_upgrade::params::mcoserver:
       ensure => file,
-      source => '/etc/mcollective/server.cfg',
+      source => $::agent_upgrade::params::oldmcoserver,
     }
   }
 
-  # Install PC1 yum repo; based off puppetlabs_yum
-  contain '::agent_upgrade::puppetlabs_yum'
+  # PLATFORM SPECIFIC CONFIGURATION
+  # Break out the platform-specific configuration into subclasses, dependent on
+  # the osfamily of the client being configured.
+
+  case $::osfamily {
+    'redhat', 'debian', 'windows', 'solaris', 'aix', 'suse': {
+      contain downcase("::agent_upgrade::osfamily::${::osfamily}")
```

Comment:
Is this a common module paradigm for "contain"ing platform specific functionality?